docs/proposals/mqtt.md

```diff
@@ -0,0 +1,234 @@
+message, and sending these messages to respective subscriber clients<sup>1</sup>. It also provides session management between
+clients and maintaining persistency for these sessions as well as authorization and authentication of clients.
+
+There are a lot of [Open-Source and Proprietary MQTT Brokers](https://mqtt.org/software/) available. One of the renowned MQTT
+Brokers is [Eclipse Mosquitto](https://mosquitto.org) which can be deployed on various infrastructure - on a single board computer
+to on-premise server to a cloud.
+
+### Message Exchange in MQTT
+
+MQTT is a Pub/Sub mechanism for message exchange. This implies that there is _never a direct connection between MQTT Clients_.
+All the message exchange occurs via mediating entity i.e. MQTT Broker. This provides a decoupling between publishing clients 
+and subscribing clients.
+
+MQTT Brokers filter incoming messages based on *Subject-Based Filtering*<sup>2</sup>. The protocol specifies the usage of 
+[MQTT Topics](https://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices/). 
+
+### MQTT Topics
+
+Topics are UTF-8 Strings in a heirarchical topic structure. Each topic level is separated by a forward slash (`/`). Example
+
+    - acme/factory/Robot0001/joint
+    - acme/factory/Machine10239/temperature
+    - system
+    - foo/bar/fizz/buzz
+
+Topics are case sensitive and must contain at least 1 character.
+
+Interested subscriber clients can either subscribe to the exact MQTT topic or can use wildcards to subscribe to multiple topics.
+
+#### MQTT Topic Wildcards
+
+1. Single Level `+`
+
+        acme/factory/+/joint
+
+    should provide information from all publishers that publish joint information
+
+2. Mutli Level `#`
+
+        acme/factory/#
+    
+    should provide information from all publishers in the acme factory
+
+## Basic Communication 
+
+![MQTT basic Communication Example](../media/akri_mqtt_basic_comm.png)
+
+Here `Device1` publishes temperature data to an MQTT Broker and `App1` and `App2` are interested in the data so they subscribe to the
+topic `device1/temp`.
+
+> Every MQTT PUBLISH message always contains the MQTT Topic string and the Payload (byte-format)
+
+A more in-depth understanding of what **MQTT PUBLISH** and **MQTT SUBSCRIBE** messages contain can be found
+[here](https://www.hivemq.com/blog/mqtt-essentials-part2-publish-subscribe/).
+
+
+> NOTE: MQTT is completely data-agnostic. The Payload and Topic strings can be developed according to the application developer's
+        perspective.
+
+
+## Discovery Process
+
+Since MQTT bases itself over the TCP/IP stack for networking, the specifications provide _handshaking_ mechanisms which can be leveraged
+to discover IoT Devices that communicate with the MQTT Broker. Since the only mediating entity for message exchange is the MQTT Broker,
+all client only require the IP Address and the MQTT port as a basic form of initiation of communication. Default port for unsecure MQTT
```

Comment:
nit: all clients
Also, maybe clarify that the clients require the "IP address and MQTT port **of the broker** as a ..."